modules/c++/mem/tests/ScratchVisualization.cpp

```diff
@@ -0,0 +1,477 @@
+#include <mem/ScratchMemory.h>
+#include <mem/BufferView.h>
+#include <cli/ArgumentParser.h>
+#include <str/Convert.h>
+#include <vector>
+#include <fstream>
+#include <string>
+
+struct Operation
+{
+    Operation(const std::string op,
+              const std::string name,
+              const size_t bytes) :
+        op(op),
+        name(name),
+        bytes(bytes)
+    {
+    }
+
+    std::string op;
+    std::string name;
+    size_t bytes;
+    mem::BufferView<sys::ubyte> buffer;
+};
+
+/*!
+ * \class Visualizer
+ *
+ * \brief Handles the inclusion and visualization of a put or release operation
+ */
+class Visualizer
+{
+public:
+    /*!
+     * Constructor
+     *
+     * \param prevOperations All operations handled in previous iterations
+     * \param iteration Which iteration is being processed
+     * \param[in,out] htmlFile Ofstream to .html file
+     * \param[in,out] cssFile Ofstream to .css file
+     */
+    Visualizer(const std::vector<Operation> prevOperations,
+               const size_t iteration,
```

Comment:
Remove const from iteration. We don't use const when passing parameters by value.